docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_global_services_enabled_remediation.mdx

```diff
@@ -1,6 +1,219 @@
+   - If you already have a trail, select it from the list of trails.
+   - If you do not have a trail, click on "Create trail" to set up a new one.
+
+3. **Configure Trail Settings:**
+   - In the trail settings, ensure that the "Management events" section is expanded.
+   - Check the box for "Read/Write events" to include both read and write management events.
+
+4. **Enable Global Events:**
+   - Scroll down to the "Advanced" section.
+   - Ensure that the option "Include global services events" is checked. This ensures that global service events (like IAM) are recorded in addition to regional events.
+
+By following these steps, you can ensure that your CloudTrail configuration captures both regional and global events, providing comprehensive logging for your AWS environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that AWS CloudTrail records both regional and global events using the AWS CLI, follow these steps:
+
+1. **Create or Update a CloudTrail Trail:**
+   Ensure you have a CloudTrail trail that is configured to log both management and data events. If you don't have a trail, create one. If you already have a trail, you can update it.
+
+   ```sh
+   aws cloudtrail create-trail --name my-trail --s3-bucket-name my-bucket
+   ```
+
+2. **Enable Logging for Global Services:**
+   Update the trail to include global service events (e.g., IAM, STS, CloudFront).
+
+   ```sh
+   aws cloudtrail update-trail --name my-trail --include-global-service-events
+   ```
+
+3. **Enable Multi-Region Trail:**
+   Ensure the trail is configured to log events from all regions.
+
+   ```sh
+   aws cloudtrail update-trail --name my-trail --is-multi-region-trail
+   ```
+
+4. **Start Logging:**
+   Start logging for the trail to ensure that events are being recorded.
+
+   ```sh
+   aws cloudtrail start-logging --name my-trail
+   ```
+
+By following these steps, you can ensure that your CloudTrail is configured to record both regional and global events.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that AWS CloudTrail records both regional and global events using Python scripts, you can use the `boto3` library, which is the AWS SDK for Python. Here are the steps to achieve this:
+
+1. **Install Boto3 Library**:
+   Ensure you have the `boto3` library installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Initialize Boto3 Client**:
+   Initialize the `boto3` client for CloudTrail.
+
+3. **Create or Update CloudTrail**:
+   Create a new CloudTrail or update an existing one to ensure it records both regional and global events.
+
+4. **Verify Configuration**:
+   Verify that the CloudTrail configuration includes both regional and global events.
+
+Here is a Python script to achieve this:
+
+```python
+import boto3
+
+# Initialize a session using Amazon CloudTrail
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>